portals/publisher/source/src/app/components/Apis/Details/Policies/AttachedPolicyCard.tsx

```diff
@@ -116,41 +120,61 @@ const AttachedPolicyCard: FC<AttachedPolicyCardProps> = ({
         event.preventDefault();
     };
 
-    const handlePolicyDownload = (event: any) => {
+    /**
+     * Handle policy download
+     * @param {React.MouseEvent<HTMLButtonElement, MouseEvent>} event event
+     */
+    const handlePolicyDownload = (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
         event.stopPropagation();
         event.preventDefault();
-        const commonPolicyContentPromise = API.getCommonOperationPolicyContent(
-            policyObj.id,
-        );
-        commonPolicyContentPromise
-            .then((commonPolicyResponse) => {
-                Utils.forceDownload(commonPolicyResponse);
-            })
-            .catch(() => {
-                const apiPolicyContentPromise = API.getOperationPolicyContent(
-                    policyObj.id,
-                    api.id,
-                );
-                apiPolicyContentPromise
-                    .then((apiPolicyResponse) => {
-                        Utils.forceDownload(apiPolicyResponse);
-                    })
-                    .catch((error) => {
-                        if (process.env.NODE_ENV !== 'production') {
-                            console.error(error);
-                            Alert.error(
-                                <FormattedMessage
-                                    id='Apis.Details.Policies.AttachedPolicyCard.download.error'
-                                    defaultMessage='Something went wrong while downloading the policy'
-                                />,
-                            );
-                        }
-                    });
-            });
+        if (policyObj.isAPISpecific) {
+            const apiPolicyContentPromise = API.getOperationPolicyContent(
+                policyObj.id,
+                api.id,
+            );
+            apiPolicyContentPromise
+                .then((apiPolicyResponse) => {
+                    Utils.forceDownload(apiPolicyResponse);
+                })
+                .catch((error) => {
+                    if (process.env.NODE_ENV !== 'production') {
+                        console.error(error);
+                        Alert.error(
+                            <FormattedMessage
+                                id='Apis.Details.Policies.AttachedPolicyCard.apiSpecificPolicy.download.error'
+                                defaultMessage='Something went wrong while downloading the policy'
+                            />,
+                        );
+                    }
+                });
+        } else {
+            const commonPolicyContentPromise = API.getCommonOperationPolicyContent(
+                policyObj.id,
+            );
+            commonPolicyContentPromise
+                .then((commonPolicyResponse) => {
+                    Utils.forceDownload(commonPolicyResponse);
+                })
+                .catch((error) => {
+                    if (process.env.NODE_ENV !== 'production') {
```

Comment:
this check is not useful IMO

Reply:
Fixed with https://github.com/wso2/apim-apps/pull/204/commits/6abcbe1f007992cc0074fbb0ba5353b67dd5f7ad